app/screens/ChatbotScreen.tsx

```diff
@@ -0,0 +1,361 @@
+  const [answer, setAnswer] = useState<ChatbotAnswer|undefined>();
+  const [progress, setProgress] = useState<number>(0);
+  const [isFinished, setFinished] = useState<boolean>(false);
+
+  useEffect(() => {
+    let id = extractSessionId(sessionUrl)
+    setSessionId(id)
+    getAuthenticationInfo(id)
+      .then(() => {
+        requestSMS(id)
+      })
+      .catch(onLoadFailed)
+  }, [])
+
+  const extractSessionId = (sessionUrl: string): string => {
+    let sessionId = sessionUrl.replace("https://services.eurospider.com/chatbot-ui/program/kyc-onboarding/", "")
+    return sessionId.split("?")[0]
+  }
+
+  const onLoadFailed = () => {
+    NotificationService.error(t("feedback.load_failed"));
+    nav.navigate(Routes.Home);
+  }
+
+  const onBack = () => {
+    // cancel
+    if (pageIndex === 0) {
+      nav.navigate(Routes.Home)
+    }
+    // back
+    else {
+      let newPageIndex = pageIndex - 1
+      setPageIndex(newPageIndex)
+      updateProgress(pages, newPageIndex, true)
+      setAnswer(pages[newPageIndex].answer)
+    }
+  }
+
+  const onNext = (pages: ChatbotPage[], isFinished: boolean, inputAnswer?: ChatbotAnswer) => {
+    if (inputAnswer === undefined) {
+      inputAnswer = answer
+    }
+    if (isFinished) {
+      setFinished(true)
+      onFinish()
+      setPageIndex(pageIndex + 1)
+      setProgress(1)
+    } else {
+      // check if we need to request next step
+      if (inputAnswer !== undefined && chatbotShouldSendAnswer(inputAnswer)) {
+        if (chatbotIsEdit(inputAnswer)) {
+          requestEditAnswer(inputAnswer.value, inputAnswer.timestamp, chatbotId)
+        } else {
+          requestNextStep(chatbotCreateAnswer(inputAnswer.value, inputAnswer), answer, chatbotId)
+        }
+      } else if (pageIndex + 1 < pages.length) {
+        let newPageIndex = pageIndex + 1
+        setPageIndex(newPageIndex)
+        setAnswer(pages[newPageIndex].answer)
+        updateProgress(pages, newPageIndex)
+      }
+    }
+  }
+
+  const requestSMS = (id?: string) => {
+    if (id !== undefined) {
+      requestSMSCode(id)
+    } else {
+      NotificationService.error(t("kyc.bot.error.missing_session_id"))
+    }
+  }
+
+  const submitSMSCode = (id?: string) => {
+    if (id !== undefined) {
+      setLoading(true)
+      postSMSCode(id, smsCode)
+        .then((id) => {
+          if (id === "false") {
+            NotificationService.error(t("kyc.bot.error.wrong_sms"))
+            // wait a short amount of time to avoid screen flickering due to fast network request
+            setTimeout(() => setLoading(false), 1000)
+          } else {
+            setChatbotId(id)
+            setSMSCompleted(true)
+            requestStart(sessionId, id)
+          }
+        })
+    } else {
+      NotificationService.error(t("kyc.bot.error.missing_session_id"))
+    }
+  }
+
+  const requestStart = (id?: string, chatbotId?: string) => {
+    getStatus(id, chatbotId)
+      .then((status) => {
+        switch (status) {
```

Comment:
what's the default case?

Reply:
good catch. Will add default => fresh start